rust/protocol/src/sealed_sender.rs

```diff
@@ -879,8 +1019,214 @@ mod sealed_sender_v2 {
 
         HKDF::new(3)?.derive_secrets(&agreement_key_input, LABEL_DH_S, AUTH_TAG_LEN)
     }
+
+    #[test]
+    fn test_agreement_and_authentication() -> Result<()> {
+        // The sender and recipient each have a long-term identity key pair.
+        let sender_identity = IdentityKeyPair::generate(&mut rand::thread_rng());
+        let recipient_identity = IdentityKeyPair::generate(&mut rand::thread_rng());
+
+        // Generate random bytes used for our multi-recipient encoding scheme.
+        let m: [u8; MESSAGE_KEY_LEN] = (&mut rand::thread_rng()).gen();
```

Comment:
`rand::thread_rng().gen()` should work, `&` or `&mut` are implicitly added for method calls even for return values.

Reply:
Thanks for the explanation! That's correct!